distribution/src/bin/elasticsearch-env.bat

```diff
@@ -17,16 +17,22 @@ rem now set the classpath
 set ES_CLASSPATH=!ES_HOME!\lib\*
 
 rem now set the path to java
-if defined JAVA_HOME (
-  set JAVA="%JAVA_HOME%\bin\java.exe"
-) else (
-  set JAVA="%ES_HOME%\jdk\bin\java.exe"
-  set JAVA_HOME="%ES_HOME%\jdk"
+rem pass "needsjava" arg to ensure JAVA_HOME and JAVA are set
+if not "%1" == "needsjava" (
```

Comment:
so when the `elasticsearch.bat install` is called, java is required, but we are only setting that when it `needsjava` is *not* set?

Reply:
This was a bug introduced due to renaming/reworking the logic to make it more clear what was happening. We actually need the default behavior to be setting JAVA/JAVA_HOME so that other scripts (eg elasticsearch.bat) don't need modification for this edge case.